packages/react-native/scripts/ios-prebuild/utils.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ * @format
+ * @oncall react_native
+ */
+
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * Creates a symbolic link from one path to another
+ * @param {string} fromPath - The path to the source file or directory
+ * @param {string} toPath - The path to the destination file or directory
+ * @returns {void}
+ * @throws {Error} If the source path does not exist or if the link creation fails
+ */
+function createLink(fromPath /*:string*/, toPath /*:string*/) {
+  if (!fs.existsSync(fromPath)) {
+    throw new Error(`Source path does not exist: ${fromPath}`);
+  }
+
+  // Check if the destination already exists
+  if (fs.existsSync(toPath)) {
+    return;
+  }
+
+  // Ensure the parent path of the link path
+  createFolderIfNotExists(path.dirname(toPath));
+
+  // Create the symbolic link
+  fs.symlinkSync(fromPath, toPath, 'dir');
```

Comment:
I think we should use `linkSync`. Xcode does not work with symlink at all. 
src: https://tlduck.com/blog/symlinks-in-xcode/

Reply:
It seems like clang is all good with using them for include files like I'm doing. Changing the script to use `linkSync` would also require us the enumerate all files - but that could be good since we could make sure to only include header files - so I've updated this as well :)